File: java/src/main/java/problems/Problem063CountingSortSimple.java
package problems;


public class Problem063CountingSortSimple {
    public static int[] sort(int[] a, int maxVal){
        int[] count=new int[maxVal+1];
        for(int x: a) count[x]++;
        int idx=0;
        for(int v=0; v<=maxVal; v++) while(count[v]-- > 0) a[idx++]=v;
        return a;
    }
}

File: scala/src/main/scala/problems/Problem063CountingSortSimple.scala

package problems

object Problem063CountingSortSimple {
  def sort(a:Array[Int], maxVal:Int): Array[Int] = {
    val count = Array.fill(maxVal+1)(0)
    a.foreach(x => count(x) += 1)
    var idx=0; var v=0
    while (v<=maxVal) { while (count(v) > 0) { a(idx)=v; idx+=1; count(v)-=1 }; v+=1 }
    a
  }
}

